Result/Result.swift

```diff
@@ -217,13 +217,19 @@ public func >>- <T, U, Error> (result: Result<T, Error>, @noescape transform: T
 // MARK: - ErrorTypeConvertible conformance
 
 #if !os(Linux)
-
-/// Make NSError conform to ErrorTypeConvertible
-extension NSError: ErrorTypeConvertible {
-	public static func errorFromErrorType(error: ErrorType) -> NSError {
-		return error as NSError
+	
+	public extension ErrorTypeConvertible where Self : NSError {
```

Comment:
Upon reflection, however, I suggest we leave this alone so as to expedite getting this PR merged.